src/test/librbd/test_librbd.cc

```diff
@@ -6765,6 +6765,106 @@ TEST_F(TestLibRBD, NamespacesPP) {
   ASSERT_EQ("name3", names[0]);
 }
 
+TEST_F(TestLibRBD, TestGetAccessTimestamp)
+{
+  REQUIRE_FORMAT_V2();
+
+  rados_ioctx_t ioctx;
+  rados_ioctx_create(_cluster, m_pool_name.c_str(), &ioctx);
+
+  rbd_image_t image;
+  int order = 0;
+  std::string name = get_temp_image_name();
+  uint64_t size = 2 << 20;  
+  struct timespec timestamp;
+  time_t ts_start;
+
+  ASSERT_EQ(0, create_image(ioctx, name.c_str(), size, &order));
+  ASSERT_EQ(0, rbd_open(ioctx, name.c_str(), &image, NULL));
+
+  ASSERT_EQ(0, rbd_get_access_timestamp(image, &timestamp));
+  ASSERT_LT(0, timestamp.tv_sec);
+
+  ts_start = timestamp.tv_sec;
+
+  // Throttling
+  sleep(65);
```

Comment:
@dillaman Still got a question by the way, I'm using a sleep() call in order to properly deal with the default throttling in the test. 
Do you think it could be relevant to add a test mid-way (e.g. after 30s) to check that the timestamp is indeed not changed and then continue as implemented ?

Reply:
Actually, now that you bring it up, that sleep statement really needs to go since it will more than double the test runtime. The best way to test that throttling is working is to use a mocked test case since you can verify what methods are being invoked on outside classes w/o the need to inject arbitrary delays. See ```test_mock_ImageRequest.cc```

Reply:
Okay, seems good. I will have a look then - thanks for the pointers